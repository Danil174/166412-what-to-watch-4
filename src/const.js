export const TextMovieRatings = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const SmallCardVideoSettings = {
  WIDTH: `100%`,
  HEIGHT: `100%`,
  IS_MUTED: true,
};

export const DEFAULT_GENRE = `All genres`;
export const MAX_GENRES_LENGTH = 9;
export const INITTIAL_LIST_LENGTH = 8;
export const SIMILAR_LIST_LENGTH = 4;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const smallCardVideoStartTimeOut = 1000;

export const LoadErrorsTexts = {
  LIST_FAIL: `Не удалось загрузить список фильмов.`,
  PROMO_FAIL: `Не удалось загрузить промо фильм.`,
};

export const AppRoute = {
  LOGIN: `/login`,
  MOVIE_PAGE: `/movie_page`,
  FAVORITES: `/favorites`,
  ROOT: `/`,
};
