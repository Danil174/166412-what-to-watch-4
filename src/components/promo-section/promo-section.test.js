import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../const.js";
import PromoSection from "./promo-section.jsx";

const mockStore = configureStore([]);

const film = {
  genre: `Crime`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  releaseDate: 2002,
  title: `Gangs of new york`,
};

describe(`PromoSectionTest`, () => {
  it(`render PromoSection`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        userData: {avatarUrl: `test.jpg`},
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <PromoSection
              film={film}
            />
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
