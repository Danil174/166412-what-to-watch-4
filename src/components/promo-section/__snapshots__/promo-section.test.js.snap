// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoSectionTest render PromoSection 1`] = `
<div
  className="movie-card__wrap"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster"
    >
      <img
        alt="Gangs of new york poster"
        height="327"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <h2
        className="movie-card__title"
      >
        Gangs of new york
      </h2>
      <p
        className="movie-card__meta"
      >
        <span
          className="movie-card__genre"
        >
          Crime
        </span>
        <span
          className="movie-card__year"
        >
          2002
        </span>
      </p>
      <div
        className="movie-card__buttons"
      >
        <button
          className="btn btn--play movie-card__button"
          type="button"
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <button
          className="btn btn--list movie-card__button"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 19 20"
            width="19"
          >
            <use
              xlinkHref="#add"
            />
          </svg>
          <span>
            My list
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
